crates/deltalake-core/src/logstore/mod.rs

```diff
@@ -0,0 +1,340 @@
+    async fn read_commit_entry(&self, version: i64) -> DeltaResult<Bytes>;
+
+    /// Write list of actions as delta commit entry for given version.
+    ///
+    /// This operation can be retried with a higher version in case the write
+    /// fails with [`TransactionError::VersionAlreadyExists`].
+    async fn write_commit_entry(
+        &self,
+        version: i64,
+        tmp_commit: &Path,
+    ) -> Result<(), TransactionError>;
+
+    /// Find latest version currently stored in the delta log.
+    async fn get_latest_version(&self, start_version: i64) -> DeltaResult<i64>;
+
+    /// Get underlying object store.
+    fn object_store(&self) -> Arc<dyn ObjectStore>;
+
+    /// [Path] to Delta log
+    fn to_uri(&self, location: &Path) -> String;
+
+    /// Get fully qualified uri for table root
+    fn root_uri(&self) -> String {
+        self.to_uri(&Path::from(""))
+    }
+
+    /// [Path] to Delta log
+    fn log_path(&self) -> &Path {
+        &DELTA_LOG_PATH
+    }
+
+    /// Check if the location is a delta table location
+    async fn is_delta_table_location(&self) -> ObjectStoreResult<bool> {
+        // TODO We should really be using HEAD here, but this fails in windows tests
+        let object_store = self.object_store();
+        let mut stream = object_store.list(Some(self.log_path())).await?;
+        if let Some(res) = stream.next().await {
+            match res {
+                Ok(_) => Ok(true),
+                Err(ObjectStoreError::NotFound { .. }) => Ok(false),
+                Err(err) => Err(err),
+            }
+        } else {
+            Ok(false)
+        }
+    }
+
+    #[cfg(feature = "datafusion")]
+    /// Generate a unique enough url to identify the store in datafusion.
+    /// The DF object store registry only cares about the scheme and the host of the url for
+    /// registering/fetching. In our case the scheme is hard-coded to "delta-rs", so to get a unique
+    /// host we convert the location from this `LogStore` to a valid name, combining the
+    /// original scheme, host and path with invalid characters replaced.
+    fn object_store_url(&self) -> ObjectStoreUrl;
+
+    /// Deletes object by `paths`.
+    async fn delete_batch(&self, paths: &[Path]) -> ObjectStoreResult<()> {
+        let object_store = self.object_store();
+        for path in paths {
+            match object_store.delete(path).await {
+                Ok(_) => continue,
+                Err(ObjectStoreError::NotFound { .. }) => continue,
+                Err(e) => return Err(e),
+            }
+        }
+        Ok(())
+    }
+
+    /// Get configuration representing configured log store.
+    fn config(&self) -> &LogStoreConfig;
+}
+
+// TODO: maybe a bit of a hack, required to `#[derive(Debug)]` for the operation builders
+impl std::fmt::Debug for dyn LogStore + '_ {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.object_store().fmt(f)
```

Comment:
Maybe it makes sense to print the root url here as well? something like
```suggestion
        write!("LogStore({})", self.root_url())
```